### PR TITLE
Add support for new stitching API fields

### PR DESCRIPTION
**Pull request by linusnorton**

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/em/orchestrator/exampleservice/ExampleBundlePopulator.java [hunk 1]
```diff
@@ -23,6 +23,9 @@ public ExampleBundlePopulator(ObjectMapper objectMapper) {
     public JsonNode populateNewBundle(JsonNode caseData) {
         CcdBundleDTO ccdBundleDTO = new CcdBundleDTO();
         ccdBundleDTO.setTitle("New Bundle");
+        ccdBundleDTO.setFileName("a-stitched-document.pdf");
```
Comment (JeroenRijks) [inline src/main/java/uk/gov/hmcts/reform/em/orchestrator/exampleservice/ExampleBundlePopulator.java]: Do we want to assign a default, or leave it blank? If we give a default, I think it should be something that will help services realise that they should make it service-specific, like "exampleservice-bundle.pdf"

File: src/main/java/uk/gov/hmcts/reform/em/orchestrator/exampleservice/ExampleBundlePopulator.java [hunk 2]
```diff
@@ -23,6 +23,9 @@ public ExampleBundlePopulator(ObjectMapper objectMapper) {
...
+        ccdBundleDTO.setHasCoversheets(true);
+        ccdBundleDTO.setHasTableOfContents(true);
```
Comment (JeroenRijks) [inline src/main/java/uk/gov/hmcts/reform/em/orchestrator/exampleservice/ExampleBundlePopulator.java]: defaults here and for coversheets are good :+1:

File: src/test/java/uk/gov/hmcts/reform/em/orchestrator/stitching/StitchingServiceTest.java [hunk 3]
```diff
@@ -41,6 +41,24 @@ public void stitchSuccessful() throws StitchingServiceException, InterruptedExce
...
+    }
+
+    @Test
+    public void stitchSuccessfulWithFileName() throws StitchingServiceException, InterruptedException {
+        List<String> responses = new ArrayList<>();
+        responses.add("{ id: 1, taskState: 'NEW', bundle: { stitchedDocumentURI: null } }");
+        responses.add("{ id: 1, taskState: 'NEW', bundle: { stitchedDocumentURI: null } }");
+        responses.add("{ id: 1, taskState: 'NEW', bundle: { stitchedDocumentURI: null } }");
+        responses.add("{ id: 1, taskState: 'NEW', bundle: { stitchedDocumentURI: null } }");
+        responses.add("{ id: 1, taskState: 'DONE', bundle: { stitchedDocumentURI: 'AAAAAA', fileName: 'a.pdf' } }");
```
Comment (JeroenRijks) [inline src/test/java/uk/gov/hmcts/reform/em/orchestrator/stitching/StitchingServiceTest.java]: why are there 5 lines here? should it only look at the latest one? if so, why isn't that in a separate test?

Comment (linusnorton) [reply src/test/java/uk/gov/hmcts/reform/em/orchestrator/stitching/StitchingServiceTest.java]: replicates the polling that happens with the stitching API

**Pull request merged**